docs/testing-with-codeception.md

```diff
@@ -0,0 +1,746 @@
+
+[Read more on WPWebDriver module configuration](https://wpbrowser.wptestkit.dev/modules/wpwebdriver#configuration).
+
+#### WPDb
+
+This module extends the `Db` module adding WordPress-specific configuration parameters and methods. It provides methods to read, write and update the WordPress database directly, without relying on WordPress methods, using WordPress functions or triggering WordPress filters.
+
+Altis comes with pre-prepared database contents that's imported on the fly to simulate a basic working site.
+
+**Important notes:**
+- `WPDb` imports the sample database content to a database called `test`, which is created (and later removed) on the fly.
+- Altis detects acceptance test requests (to the actual running application) and switches the database to `test` in runtime, so it doesn't mess with existing site content.
+
+[Read more on WPDb module configuration](https://wpbrowser.wptestkit.dev/modules/wpdb#configuration).
+
+#### WPLoader
+
+This module is typically used in integration tests, to bootstrap WordPress code in the context of the tests.
+It can also be used in acceptance and functional tests, by setting the `loadOnly` parameter to `true`, in order to acccess WordPress code in the tests context (using the `tests` database imported by `WPDb` ).
+
+This module is a wrapper around the functionalities provided by the [WordPress PHPUnit Core test suite](https://make.wordpress.org/core/handbook/testing/automated-testing/phpunit/), as such it provides the same methods and facilities.
+
+The parameters provided to the module duplicate the ones used in the WordPress configuration file. `WPLoader` will not bootstrap WordPress using the `wp-config.php` file, it will define and use its own WordPress configuration values passed from the defined module parameters.
+
+**Important notes:**
+
+- If `loadOnly` parameter is set to `false`, Codeception will execute all database modification requests, eg: created and/or deleted content, as a sql transaction, which gets rolled-back whenever the test scenario completes.
+- WordPress defined functions and classes (and those of the plugins and themes loaded with it) will be available in the `setUpBeforeClass` method.
+- WordPress would not have loaded yet when PHPUnit calls the data provider methods, so don't expect to be able to use any WordPress functions within data provider methods.
+
+[Read more on WPLoader module configuration](https://wpbrowser.wptestkit.dev/modules/wploader#configuration).
+
+#### WPQueries
+
+This module is typically used in integration tests, to make assertions on the database queries made by the global $wpdb object, and it **requires** `WPLoader` module to work. It will set, if not set already, the `SAVEQUERIES` constant to true and will throw an exception if the contstant is already set to a falsy value.
```

Comment:
```suggestion
This module is typically used in integration tests, to make assertions on the database queries made by the global $wpdb object, and it **requires** the `WPLoader` module in order to work. It will set, if not set already, the `SAVEQUERIES` constant to true and will throw an exception if the constant is already set to a falsy value.
```